external/llvm-project/mlir/include/mlir/Dialect/AMDGPU/IR/AMDGPU.td

```diff
@@ -379,6 +379,10 @@ def MFMAOutTypes : AnyTypeOf<[F64,
                               VectorOfLengthAndType<[4, 16, 32], [F32]>,
                               VectorOfLengthAndType<[4, 16, 32], [I32]>,
                               VectorOfLengthAndType<[4], [F64]>]>;
+// wmma
+def WMMAInTypes : AnyTypeOf<[VectorOfLengthAndType<[16], [F16, BF16, I8, SI8, UI8]>]>;
+def WMMAOutTypes : AnyTypeOf<[VectorOfLengthAndType<[4, 8], [F32, I32]>,
+                              VectorOfLengthAndType<[8, 16], [F16, BF16]>]>;
```

Comment:
[Question] If the result is always 16x16 elements, why there are two alternative vector length each? Does it have anything to do with wave32 and wave64 mode? Are we using/enabling both modes?

Reply:
Also, get rid of the `f16` as Jing mentioned that they are same bandwidth but lower precision?

Reply:
This is a backend wrapper - if the hardware can do it, we add it.